packages/tree-view/src/behaviors/TreeItemBehavior.js

```diff
@@ -69,16 +67,28 @@ export default class TreeItemBehavior extends Component {
       const index = treeItemArray !== null && treeItemArray.indexOf(id);
       setActiveTreeItemId(id);
       setActiveTreeItemIndex(index);
+    }
+  };
+
+  handleOperatorClick = (event, treeItem) => {
+    event.stopPropagation();
+
+    if (this.props.payload) {
+      // eslint-disable-next-line no-param-reassign
+      treeItem.meta.collapsed = !treeItem.meta.collapsed;
+      this.setIsCollapsed(treeItem.meta.collapsed);
```

Comment:
this is the line that sets collapsed... the `setIsCollapsed` doesn't seem to do anything.

Reply:
it does, without that line the open/close wont work properly for the tree object.